File: src/Sail.Core/Entities/Route.cs
namespace Sail.Core.Entities;

public class Route
{
    public Guid Id { get; set; }
    public Guid? ClusterId { get; set; }
    public string Name { get; set; } = string.Empty;
    public RouteMatch Match { get; set; } = new();
    public int Order { get; set; }
    public string? AuthorizationPolicy { get; set; }
    public string? RateLimiterPolicy { get; set; }
    public string? CorsPolicy { get; set; }
    public string? TimeoutPolicy { get; set; }
    public TimeSpan? Timeout { get; set; }
    public long? MaxRequestBodySize { get; set; }
    public bool? HttpsRedirect { get; set; }
    public List<Dictionary<string, string>>? Transforms { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: web/src/types/route.ts
export interface Route {
  id: string;
  name: string;
  clusterId?: string;
  order: number;
  path?: string;
  methods?: string[];
  hosts?: string[];
  enabled: boolean;
  authorizationPolicy?: string;
  rateLimiterPolicy?: string;
  corsPolicy?: string;
  timeoutPolicy?: string;
  retryPolicy?: string;
  timeout?: string;
  maxRequestBodySize?: number;
  httpsRedirect?: boolean;
  createdAt: string;
  updatedAt: string;
  match?: {
    path?: string;
    methods?: string[];
    hosts?: string[];
    headers?: Array<{
      name: string;
      values?: string[];
      mode?: number;
      isCaseSensitive?: boolean;
    }>;
    queryParameters?: Array<{
      name: string;
      values?: string[];
      mode?: number;
      isCaseSensitive?: boolean;
    }>;
  };
  transforms?: Array<Record<string, string>>;
  metadata?: Record<string, string>;
}

export type RouteStatus = 'active' | 'inactive' | 'error';

